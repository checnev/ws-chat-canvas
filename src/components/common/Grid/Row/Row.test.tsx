import { render, unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';
import pretty from 'pretty';
import GridRow from 'components/common/Grid/Row';

let container: any = null;

beforeEach(() => {
  container = document.createElement('div');
  document.body.append(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

const setUp = (props?: any) => (
  <GridRow {...props}>
    Grid Column Text
  </GridRow>
);

describe('GridRow component', () => {
  it('should render GridRow', () => {
    act(() => {
      render(setUp(), container);
    });

    expect(
      pretty(container.innerHTML)
    ).toMatchSnapshot();
  });

  it('should render GridRow with props', () => {
    const props = {
      justifyContent: 'center',
      alignItems: 'center',
      flexDirection: 'column',
    };

    act(() => {
      render(setUp(props), container);
    });

    expect(
      pretty(container.innerHTML)
    ).toMatchSnapshot();
  });
});
