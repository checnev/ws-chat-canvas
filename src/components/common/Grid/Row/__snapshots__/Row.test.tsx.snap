// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GridRow component should render GridRow 1`] = `"<div class=\\"gridRow\\">Grid Column Text</div>"`;

exports[`GridRow component should render GridRow with props 1`] = `"<div class=\\"gridRow gridRow_justifyContent_center gridRow_alignItems_center gridRow_flexDirection_column\\">Grid Column Text</div>"`;
